statistics.cpp

```diff
@@ -27,42 +29,86 @@ double Statistics::logLikelihood(const std::valarray<double>& particleSizeData,
 
   // isolate the relevant particle sizes
   std::valarray<double> particleSizes (largestParticle - smallestParticle + 1);
-  std::valarray<double> concentrations (particleSizes.size());
   for (unsigned int size = smallestParticle; size < largestParticle + 1; size++)
   {
     // vector holding particle sizes
     particleSizes[size - smallestParticle] = size;
-    // vector holding corresponding concentrations
-    concentrations[size - smallestParticle] = particleSizeDistr[odeModel.particleSizeToIndex(size, odeParameters)];
   }
 
-  // normalize subset vector so its sum is equal to 1
-  const double normalizeConst = concentrations.sum();
-  const std::valarray<double> particleSizePMF = concentrations/normalizeConst;
+  // start with log likelihood at 0
+  double logLikelihoodVal = 0.0;
 
-  // create histogram from PMF to aggregate bin probability
-  Histograms::Histogram odeHistogram(histogramParameters);
+  // instantiate vectors for future output
+  std::vector<std::valarray<double>> pmfs;
+  std::vector<std::valarray<double>> data_counts;
```

Comment:
Do you actually carry their values from one iteration of the loop below to the next iteration? If not, then move these variables' declarations into the loop, just above where you first need them -- remember the section about the lifetime of objects?